File: webclient/src/app/models/date.ts
export class Date
{
    private year:number;
    private month:number;
    private day:number;
    private hour:number;
    private minute:number;

    public constructor(year:number,month:number,day:number,hour?:number,minute?:number)
    {
        this.year = year;
        this.month = month;
        this.day = day;
        this.hour = hour;
        this.minute = minute;
    };

    public get getYear(): number {
        return this.year;
    }

    public set setYear(value: number) {
        this.year = value;
    }

    public get getMonth(): number {
        return this.month;
    }

    public set setMonth(value: number) {
        this.month = value;
    }

    public get getDay(): number {
        return this.day;
    }

    public set setDay(value: number) {
        this.day = value;
    }

    public get getHour(): number {
        return this.hour;
    }

    public set setHour(value: number) {
        this.hour = value;
    }

    public get getMinute(): number {
        return this.minute;
    }
    public set setMinute(value: number) {
        this.minute = value;
    }
}

File: src/Date.java

public class Date {
    private int day;
    private int month;
    private int year;
    private int hour;
    private int minute;

    public Date(int day, int month, int year, int hour, int minute) {
        this.day = day;
        this.month = month;
        this.year = year;
        this.hour = hour;
        this.minute = minute;
    }

    public Date(int day, int month, int year) {
        this.day = day;
        this.month = month;
        this.year = year;
        this.hour = -1;
        this.minute = -1;
    }

    public int getDay() {
        return day;
    }

    public void setDay(int day) {
        this.day = day;
    }

    public int getMonth() {
        return month;
    }

    public void setMonth(int month) {
        this.month = month;
    }

    public int getYear() {
        return year;
    }

    public void setYear(int year) {
        this.year = year;
    }

    public int getHour() {
        return hour;
    }

    public void setHour(int hour) {
        this.hour = hour;
    }

    public int getMinute() {
        return minute;
    }

    public void setMinute(int minute) {
        this.minute = minute;
    }

    @Override
    public String toString() {
        if(hour==-1 && minute==-1)
            return day+"/"+month+"/"+year;
        else
            return day+"/"+month+"/"+year+" "+hour+":"+minute;
    }


}